packages/apollos-api/src/data/features/data-source.js

```diff
@@ -77,14 +77,24 @@ export default class Feature extends baseFeature.dataSource {
     return featureListItems.filter((item) => !!item);
   }
 
-  async createFeatureId({ args, type }) {
-    const {
-      campusId,
-    } = await this.context.dataSources.Person.getCurrentUserCampusId();
-    // doing this helps ensure that features are different because of different campuses end up with different ids.
-    return createGlobalId(JSON.stringify({ campusId, ...args }), type);
+  // add in campus to feature IDs
+  createFeatureId({ args }) {
+    console.log(this.context.campusId);
+    return JSON.stringify({
+      campusId: this.context.campusId || null,
+      ...args,
+    });
   }
 
+  // pop out campus ID from feature feed id
+  baseGetFeatures = this.getFeatures;
+
+  getFeatures = async (args) => {
+    // campusId is only in the args from our custom FeatureFeed.getFeed function
+    this.context.campusId = args.campusId;
```

Comment:
and then it was getting loaded in here

Reply:
The campusId `arg` isn't passed through I think. `args` is an array of features (the config). We need to change this is core so the second arg of `getFeatures` is the `args` for the FeatureFeed

Reply:
oh I see. I meant to embed campus id one level deeper on each feature config